packages/bot/src/util/handleStaffThreadMessage.ts

```diff
@@ -0,0 +1,88 @@
+import { PrismaClient } from '@prisma/client';
+import type { ChatInputCommandInteraction, ThreadChannel } from 'discord.js';
+import i18next from 'i18next';
+import { container } from 'tsyringe';
+import { sendStaffThreadMessage, type SendStaffThreadMessageOptions } from './sendStaffThreadMessage';
+
+/**
+ *
+ * @param interaction A received interaction from the edit and reply commands.
+ * @param action Which command was used to call this function.
+ */
+export async function handleStaffThreadMessage(
+	interaction: ChatInputCommandInteraction<'cached'>,
+	action: 'reply' | 'edit',
+) {
+	const prisma = container.resolve(PrismaClient);
+
+	const thread = await prisma.thread.findFirst({
+		where: { channelId: interaction.channelId, closedById: null },
+	});
+	if (!thread) {
+		return interaction.reply(i18next.t('common.errors.no_thread'));
+	}
+
+	const options: Partial<SendStaffThreadMessageOptions> = {
+		content: interaction.options.getString('content', true),
+		staff: interaction.member,
+		channel: interaction.channel as ThreadChannel,
+		interaction,
+		threadId: thread.threadId,
+	};
+
+	const attachment = interaction.options.getAttachment('attachment');
+
+	const member = await interaction.guild.members.fetch(thread.userId).catch(() => null);
+	if (!member) {
+		return interaction.reply(i18next.t('common.errors.no_member', { lng: interaction.locale }));
+	}
+	options.member = member;
+
+	if (action === 'reply') {
+		options.anon = interaction.options.getBoolean('anon') ?? false;
+		options.attachment = attachment;
+	} else {
+		const id = interaction.options.getInteger('id', true);
+		const threadMessage = await prisma.threadMessage.findFirst({ where: { thread, localThreadMessageId: id } });
+		if (!threadMessage) {
+			return interaction.reply(
+				i18next.t('common.errors.resource_not_found', { resource: 'message', lng: interaction.locale }),
+			);
+		}
+
+		if (threadMessage.staffId !== interaction.user.id) {
+			return interaction.reply(i18next.t('common.errors.not_own_message', { lng: interaction.locale }));
+		}
+
+		const clearAttachment = interaction.options.getBoolean('clear-attachment');
+
+		if (attachment && clearAttachment) {
+			return interaction.reply(
+				i18next.t('common.errors.arg_conflict', {
+					first: 'attachment',
+					second: 'clear-attachment',
+					lng: interaction.locale,
+				}),
+			);
+		}
+
+		const guildMessage = await interaction.channel!.messages.fetch(threadMessage.guildMessageId);
+		const userChannel = await member.createDM();
+		const userMessage = await userChannel.messages.fetch(threadMessage.userMessageId);
+
+		Object.assign<typeof options, Partial<typeof options>>(options, {
```

Comment:
Frankly I think I'd like to avoid `Object.assign` unless we absolutely need to use it (e.g. if we get hackier types otherwise). Is there anything stopping you from just doing object spreading here?